client/rest-high-level/src/main/java/org/elasticsearch/client/transform/transforms/latest/LatestDocConfig.java

```diff
@@ -0,0 +1,134 @@
+
+/**
+ * Class describing how to compute latest doc for every unique key
+ */
+public class LatestDocConfig implements ToXContentObject {
+
+    private static final ParseField UNIQUE_KEY = new ParseField("unique_key");
+    private static final ParseField SORT = new ParseField("sort");
+
+    private final List<String> uniqueKey;
+    private final List<SortBuilder<?>> sort;
+
+    @SuppressWarnings("unchecked")
+    private static final ConstructingObjectParser<LatestDocConfig, Void> PARSER =
+        new ConstructingObjectParser<>(
+            "latest_doc_config", true, args -> new LatestDocConfig((List<String>) args[0], (List<SortBuilder<?>>) args[1]));
+
+    static {
+        PARSER.declareStringArray(constructorArg(), UNIQUE_KEY);
+        PARSER.declareField(constructorArg(), (p, c) -> SortBuilder.fromXContent(p), SORT, ObjectParser.ValueType.OBJECT_ARRAY_OR_STRING);
+    }
+
+    public static LatestDocConfig fromXContent(final XContentParser parser) {
+        return PARSER.apply(parser, null);
+    }
+
+    LatestDocConfig(List<String> uniqueKey, List<SortBuilder<?>> sort) {
+        this.uniqueKey = uniqueKey;
+        this.sort = sort;
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject();
+        builder.field(UNIQUE_KEY.getPreferredName(), uniqueKey);
+        builder.field(SORT.getPreferredName(), sort);
+        builder.endObject();
+        return builder;
+    }
+
+    public List<String> getUniqueKey() {
+        return uniqueKey;
+    }
+
+    public List<SortBuilder<?>> getSort() {
+        return sort;
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        if (this == other) {
+            return true;
+        }
+        if (other == null || getClass() != other.getClass()) {
+            return false;
+        }
+        LatestDocConfig that = (LatestDocConfig) other;
+        return Objects.equals(this.uniqueKey, that.uniqueKey) && Objects.equals(this.sort, that.sort);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(uniqueKey, sort);
+    }
+
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    public static class Builder {
+        private List<String> uniqueKey;
+        private List<SortBuilder<?>> sort;
+
+        /**
+         * Set how to group the source data
+         * @param uniqueKey The configuration describing how to group the source data
+         * @return the {@link Builder} with the interval set.
+         */
+        public Builder setUniqueKey(String... uniqueKey) {
+            return setUniqueKey(Arrays.asList(uniqueKey));
+        }
+
+        public Builder setUniqueKey(List<String> uniqueKey) {
+            this.uniqueKey = uniqueKey;
+            return this;
+        }
+
+        public Builder setSort(FieldSortBuilder sort) {
```

Comment:
> do we support all options? This seems problematic for continuous. Maybe we should start simple and only allow strings?

My intention is to allow strings rather than (in addition to) full sort spec. Let me verify that but IIUC an array of strings will be parsed correctly by this line:
```
        PARSER.declareField(constructorArg(), (p, c) -> SortBuilder.fromXContent(p), SORT, ObjectParser.ValueType.OBJECT_ARRAY_OR_STRING);
```

In the fullness of time I think we should support both `ASC` and  `DESC` as `latest` could work for any monotonic function (either increasing or decreasing).
But in the first version we can prevent that and require that the only sort field is `SyncConfig.timestamp` and always use `order` of `DESC`